### DISCUSSION TITLE
Allow drag and drop into blank space to start new line

### DISCUSSION THREAD

**Discussion by advice2020** [geany/geany#4059] [Ideas]
I noticed that if try to drag and drop text into the blank space (area that does not contain line numbers yet) that it generates some undesired results.
Here are a couple of simple examples.

`EXAMPLE 1`
If you have a document that looks like the following
1 apple
2 orange

Default Behavior:
If you either copy text and paste OR drag and drop text from elsewhere into the blank space at bottom of this document
banana
it ends up putting that line of text at the end of the last line number
1 apple
2 orangebanana

Expected Result:
1 apple
2 orange
3 banana

`EXAMPLE 2`
If you do the same as the example 1 above but instead have multiple lines of text, it does the following
banana
grape
pear

Default Behavior:
It adds that first line to the end of that line number and then creates new line numbers for the other lines
1 apple
2 orangebanana
3 grape
4 pear

Expected Result:
1 apple
2 orange
3 banana
4 grape
5 pear

------------------------------------------------------------------------------------------------

Drag and drop is based on where the cursor is located.
So when you perform this actions in the blank area below, the cursor gets moved to the end of the last line numbers text, causing this undesired result.
I guess I am suggesting to add an option/setting that when text is dragged and dropped into blank space, that it automatically creates a new line, instead of adding it to the end of an existing line, if anyone is interested in adding something like this.
Note:
This kind of mimics similar scenario with file managers where if the pane contains many files/lines that no more blank space is available.

Thank You

**Comment by elextr** [comment]
Some users want patch same location, others want new line, no solution,
same line is history and wins

**Comment by advice2020** [comment]
This would require something similar to "virtual spaces" to be added vertically.
I made a similar post about this found here 4379 (probably not worth reading)
Scintilla developer was not interested in the idea so I closed the post.
So I will also close this post as well.
